### PR TITLE
Fix compatibility with docker 1.12

### PR DESCRIPTION
**Pull request by iamluc**
Can be tested with

```
docker run -d --name docker-hostmanager -v /var/run/docker.sock:/var/run/docker.sock -v /etc/hosts:/hosts --restart=always iamluc/docker-hostmanager:compat-docker-1.12
```

### CODE REVIEW

**Comment by iamluc** [comment]
ping @kevin-verschaeve

File: src/Docker/Docker.php [hunk 1]
```diff
@@ -3,18 +3,43 @@
...
+
+    public function __construct(HttpClient $httpClient = null, Serializer $serializer = null, MessageFactory $messageFactory = null)
+    {
+        $this->httpClient = $httpClient ?: DockerClient::createFromEnv();
+        $this->messageFactory = $messageFactory ?: new GuzzleMessageFactory();
+
+        parent::__construct($this->httpClient, $serializer, $this->messageFactory);
+    }
+
+        /**
```
Comment (kevin-verschaeve) [inline src/Docker/Docker.php]: indentation ;)

**Comment by kevin-verschaeve** [comment]
great ! :+1:

**Comment by iamluc** [comment]
Change target to branch develop

**Pull request closed without merging**